Execute atomic

spaghettidba opened the pull request:
Modified to execute migrations one at a time and identify errors in migration scripts more easily

gimmi committed:
Release

spaghettidba committed:
Merge branch 'master' of https://github.com/spaghettidba/sqlmigrator.git

spaghettidba committed:
Modified to execute migrations one at a time and identify  errors in migration scripts more easily

spaghettidba committed:
Fixed bug in script execution using SMO instead of SqlCommand

spaghettidba committed:
Filter out migration (-1) while building script

spaghettidba committed:
Refined script creation

gimmi commented:
Hi,

First of all, the project was not touched in the past 4 year, `master` branch and was working, so I updated the project to build on a "contemporary" development setup (VS2015). You should be able to build just fine now.

Then I switched to your PR and the build failed because of the inclusion of:

```
using Microsoft.SqlServer.Management.Common;
using Microsoft.SqlServer.Management.Smo;
```

Seems that these object are not part of the .NET framework.

spaghettidba commented:
They are part of SQL server smo. I have no idea whether there's an official
nuget package for that, I don't think so.
I used an unofficial package, but didn't know how to include it, since the
project did not build as it was (required Java to build with the included
script, and the day I'm installing Java on my workstation has yet to come).

Il 01 Apr 2017 4:50 PM, "Gian Marco Gherardi" <notifications@github.com> ha
scritto:

> Hi,
>
> First of all, the project was not touched in the past 4 year, master
> branch and was working, so I updated the project to build on a
> "contemporary" development setup (VS2015). You should be able to build just
> fine now.
>
> Then I switched to your PR and the build failed because of the inclusion
> of:
>
> using Microsoft.SqlServer.Management.Common;
> using Microsoft.SqlServer.Management.Smo;
>
> Seems that these object are not part of the .NET framework.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/gimmi/sqlmigrator/pull/4#issuecomment-290924728>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AHPbdhAwDg_0FiuH5pCv5OHuWfojjJTIks5rrmQrgaJpZM4Ml6_Y>
> .
>

gimmi commented:
If you want to put the developer hat on, you have to have Java installed :)

Anyway, can you point me to the NuGet package? I will take a look at it.

spaghettidba commented:
Haha! That hat appears on my head from time to time, but it's not really my
thing any more...
I would install Java on a throwaway VM, but never on my main work machine.
It's been going downhill since Oracle acquired it years ago.
The Nuget package I used is this one:
https://www.nuget.org/packages/Microsoft.SQLServer.SMO/

Thanks!!

Il 02 Apr 2017 08:38, "Gian Marco Gherardi" <notifications@github.com> ha
scritto:

> If you want to put the developer hat on, you have to have Java installed :)
>
> Anyway, can you point me to the NuGet package? I will take a look at it.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/gimmi/sqlmigrator/pull/4#issuecomment-290968051>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AHPbdsWu2zBvX8LyII6hnts_VnzlC-6qks5rr0J6gaJpZM4Ml6_Y>
> .
>
